src/control/common/proto/mgmt/svc.pb.go

```diff
@@ -1132,6 +1133,13 @@ func (x *GetAttachInfoResp_RankUri) GetProviderIdx() uint32 {
 	return 0
 }
 
+func (x *GetAttachInfoResp_RankUri) GetNumCtxs() uint32 {
+	if x != nil {
+		return x.NumCtxs
+	}
+	return 0
```

Comment:
i wonder if 1 is a better default. all servers can be assumed to have at least 1 context

Reply:
This is autogenerated protobuf code so we can't set defaults here unfortunately. A good point though, we should set defaults when we read the config.

Reply:
Verified in the code that we're already checking for this default case if the field is unset in the yaml. If a user explicitly specifies 0 endpoints, that is an invalid config.